# Copyright 2018 NoMagic Sp. z o.o.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

#Action goal

# DEPRECATED int32 START_PROGRAM = 0
int32 MOVE_JOINT = 1
int32 MOVE_JOINT_IK = 2
int32 MOVE_LINEAR = 3
int32 MOVE_LINEAR_FK = 4
int32 STOP_PROGRAM = 99

int32 MOVE_GRIPPER = 10
int32 MOVE_GRIPPER_DC = 11

int32 VACUUM_ON = 20
int32 VACUUM_OFF = 21

int32 WARNING_ON = 22
int32 WARNING_OFF = 23

int32 FT_ZERO = 30
int32 FT_SEARCH = 31

int32 MOVEIT = 40

int32 MOVE_TO_SUCK = 50

int32 move_type
float64[6] target_joints
float64 velocity
float64 acceleration
geometry_msgs/Pose target_pose
float64 gripper_width
float64 gripper_force

nomagic_ur_driver/ForceTorqueThreshold ft_threshold
---
int32 result
---
